examples/rotating-voronoi/src/index.ts

```diff
@@ -0,0 +1,222 @@
+import { serialize } from "@thi.ng/hiccup";
+import { canvas } from "@thi.ng/hdom-canvas";
+import { DVMesh } from "@thi.ng/geom-voronoi";
+import { simplify } from "@thi.ng/geom-resample";
+import {
+    pathFromCubics,
+    asCubic,
+    Polygon,
+    withAttribs,
+    svgDoc,
+    rect,
+    group,
+    points
+} from "@thi.ng/geom";
+import { CubicOpts } from "@thi.ng/geom-api";
+import { cartesian2, Vec } from "@thi.ng/vectors";
+import { TAU, PI } from "@thi.ng/math";
+import { convertTree } from "@thi.ng/hiccup-svg";
+import { clearDOM } from "@thi.ng/hdom";
+import { updateDOM } from "@thi.ng/transducers-hdom";
+import {
+    AppState,
+    scaleStream,
+    animationStream,
+    frameStreamConditional,
+    frameStream,
+    keyStream,
+    keyStreamConditional,
+    mainStream
+} from "./stream-state";
+import {
+    map,
+    iterator1,
+    normRange,
+    comp,
+    mapcat,
+    iterator
+} from "@thi.ng/transducers";
+import { slider, checkbox } from "./controllers";
+import { download } from "./download";
+
+const edge = window.innerWidth * 0.7;
+const width = edge;
+const height = edge;
+const radius = (width / 2) * 0.8;
+const center = [width / 2, height / 2];
+
+const rndInt = (min: number, max: number) =>
+    Math.trunc(min + Math.random() * (max - min));
+
+const startingCircles: Array<[number, number, boolean]> = [
+    [radius / 1, rndInt(4, 20), true],
+    [radius / 2, rndInt(4, 20), false],
+    [radius / 4, rndInt(4, 20), true],
+    [radius / 8, rndInt(4, 20), false]
+];
+
+const pointsInCircle = (
+    _center: Vec,
+    _radius: number,
+    _num: number,
+    _angle: number
+) => [
+    ...iterator1(
+        map((index) =>
+            cartesian2([], [_radius, index * TAU + _angle], _center)
+        ),
+        normRange(_num, false)
+    )
+];
+
+scaleStream.next(1);
+frameStreamConditional.next(0);
+animationStream.next(false);
+const startKeyEvent: KeyboardEvent = document.createEvent("KeyboardEvent");
+startKeyEvent.initEvent("keyup");
+keyStreamConditional.next(startKeyEvent);
+
+const app = (...args: any) => (state: AppState) => appRender(state);
+mainStream.transform(
+    map(app(scaleStream, animationStream, frameStream, keyStream)),
+    updateDOM()
+);
+
+function computeVoronoi(state: AppState) {
+    const delta = state.frameValue / 100;
+    const doSave = state.keyValue === "s";
+
+    const opts: CubicOpts = {
+        breakPoints: false,
+        uniform: false,
+        scale: state.scaleValue
+    };
+
+    const startPoints = [
```

Comment:
the additional `mapcat` is superfluous here, can just use it directly and avoids needing `iterator` & `comp`:

```ts
const startPoints = [
    ...mapcat(([rad, density, clockwise]) =>
        pointsInCircle(
            center,
            rad,
            density,
            clockwise ? delta : PI - delta
        ),
        startingCircles
    )
];
```